### PR TITLE
GH-39352: [FS][Azure] Enable azure in builds

### LINKED ISSUE #39352: [Python][FS][Azure] Enable Azure in pyarrow wheel builds
### Describe the enhancement requested

Child issue of https://github.com/apache/arrow/issues/39317

Currently Azure is disabled in pyarrow wheel builds. 

There are a collection of scripts for pyarrow wheel builds on different OSs e.g. `python_wheel_macos_build.sh`, `python_wheel_manylinux_build.sh`, `python_wheel_unix_test.sh`, `python_wheel_windows_build.bat`, ...

Probably we should try to enable Azure by default in all the cases where S3 and GCS are also enabled by default. 

### Component(s)

Python

### PR DESCRIPTION
**Pull request by Tom-Newton**
### What changes are included in this PR?
Enable Azure in linux and mac os wheel builds. Tried to copy GCS

Don't enable Azure for windows builds because windows builds where all failing. Failures were a combination of cmake version or `Could not find a package configuration file provided by "wil"`. I think it makes sense to come back to windows builds in another PR.  

### Are these changes tested?
There is no new functionality to test. 

### Are there any user-facing changes?
No

* Closes: #39352

### CODE REVIEW

**Comment by Tom-Newton** [comment]
I think this is ready for review. I guess we also need to run some archery builds?

File: ci/scripts/python_wheel_unix_test.sh [hunk 1]
```diff
@@ -32,11 +32,13 @@ source_dir=${1}
 : ${ARROW_SUBSTRAIT:=ON}
 : ${ARROW_S3:=ON}
 : ${ARROW_GCS:=ON}
+: ${ARROW_AZURE:=ON}
```
Comment (kou) [inline ci/scripts/python_wheel_unix_test.sh:35]: Could you sort this list in alphabetical order?
(including `ARROW_GCS`)

**Comment by kou** [comment]
@github-actions crossbow submit -g wheel

**Comment by kou** [comment]
@github-actions crossbow submit -g wheel

**Comment by Tom-Newton** [comment]
Damn. It looks like I missed the second macos vcpkg install. I think big sur builds should be fixed now

**Comment by Tom-Newton** [comment]
I think the one failure is probably a flake. The same one passed in other archery builds

**Review by kou — APPROVED** [review]
+1

Thanks!

**Pull request merged**